test/create_k8s_token_test.rb

```diff
@@ -0,0 +1,132 @@
+require_relative 'test_helper'
+require 'rack'
+require 'json'
+require 'open-uri'
+
+SingleCov.not_covered!
+
+describe 'CRON' do
+  class FakeServer
+    TEST_ENDPOINT = "/__ping__".freeze
+
+    def self.open(port, replies)
+      server = new(port, replies)
+      server.boot
+      yield server
+    ensure
+      server.shutdown
+    end
+
+    def initialize(port, replies)
+      @port = port
+      @replies = replies.merge(TEST_ENDPOINT => 'PONG')
+    end
+
+    def boot
+      Thread.new do
+        Rack::Handler::WEBrick.run(
+          self,
+          Port: @port,
+          Logger: WEBrick::Log.new("/dev/null"),
+          AccessLog: []
+        ) { |s| @server = s }
+      end
+    end
+
+    def wait
+      loop do
+        begin
+          break if URI.parse("http://localhost:#{@port}#{TEST_ENDPOINT}").read
+        rescue
+          nil
+        end
+      end
+    end
+
+    def call(env)
+      path = env.fetch("PATH_INFO")
+      unless reply = @replies[path]
+        puts "ERROR: Missing reply for path #{path}" # kubeclient does not show current url when failing
+        raise
+      end
+      [200, {}, [reply.to_json]]
+    end
+
+    def shutdown
+      @server.shutdown if @server
+    end
+  end
+
+  let(:vault_replies) do
+    {
+      '/auth/token/create' =>
+        {
+          "auth": {
+            "client_token": "ABCD",
+            "policies": ["web", "stage"],
+            "metadata": {"user": "armon"},
+            "lease_duration": 3600,
+            "renewable": true,
+          }
+        },
+      '/auth' => {}
+
+    }
+  end
+
+  let(:kuber_replies) do
+    {
+      '/api/v1/namespaces/default/secrets/vaultauth' => {},
+    }
+  end
+
+  describe "all the tests" do
+    Minitest::Test.class_eval do
+      Thread.abort_on_exception = true
+    end
+
+    around do |test|
+      WebMock.disable!
+      FakeServer.open(8200, vault_replies) do |vault|
+        FakeServer.open(8443, kuber_replies) do |kube|
+          vault.wait
+          kube.wait
+          with_env(
+            CRON_TEST: 'true',
+            VAULT_PORT: "8200",
+            BUNDLER_ROOT: Bundler.root.to_s,
+            &test
+          )
+        end
+      end
+    end
+
+    describe "failure cases" do
+      around { |t| with_env(VAULT_PORT: "8201", &t) }
```

Comment:
needed as it overrides the port that the stub'd client is using and creates the failure.